Bluepill-cli/Bluepill-cli/Bluepill/Bluepill.m

```diff
@@ -480,16 +544,38 @@ - (void)finishWithContext:(BPExecutionContext *)context {
             context.finalExitStatus = BPExitStatusAppCrashed;
             NEXT([self proceed]);
             return;
+        case BPExitStatusSimulatorDeleted:
+            self.finalExitStatus = BPExitStatusSimulatorDeleted;
+            self.exitLoop = YES;
+            return;
     }
 
 }
 
+- (void) writeDeviceIDFile {
+    NSString *idStr = [self getSimulatorDeviceID];
+    if (!idStr || !self.context.simulatorCreated) return;
+    
+    NSString *tempFileName = [NSString stringWithFormat:@"bluepill-deviceid.%d",getpid()];
+    NSString *tempFilePath = [NSTemporaryDirectory() stringByAppendingPathComponent:tempFileName];
```

Comment:
I tried the mkstemp, but it doesn't fit here, because it will append a random string to the temp file name, which the driver process doesn't know, so the driver cannot find the file to read.  Currently the only variable in the temp file name is the pid which both driver and bp know.